google-cloud-spanner/src/main/java/com/google/cloud/spanner/InstanceAdminClientImpl.java

```diff
@@ -193,9 +193,18 @@ public Operation fromProto(Operation proto) {
   @Override
   public OperationFuture<Instance, CreateInstanceMetadata> createInstance(InstanceInfo instance)
       throws SpannerException {
+    boolean valid_capacity_with_autoscaling =
+        instance.getAutoscalingConfig() != null
```

Comment:
Is this validation just specific to `createInstance` RPC? What about other RPCs ? I am wondering if these validations are better suited when we invoke `setAutoscalingConfig` method than within RPC.

WDYT?

Reply:
Also, how are such validations supported in auto-generated languages (for ex - Golang) ?

Reply:
Sure, let's get rid of the validation all together and rely on the backend validation. Some simpler validation on primitive types can be considered to be moved to the protos. This in general is preferred since we are gradually moving in a direction on how we can simplify our handwritten clients (similar to our auto-generated clients). So, this would be a step in that direction.